idx/memory/memory_test.go

```diff
@@ -279,6 +279,30 @@ func TestDelete(t *testing.T) {
 	})
 }
 
+func TestDeleteWithLotsOfSeries(t *testing.T) {
```

Comment:
the complexity of the old code was exponential, O(n^2).  But this new code is O(n)

With the old code, it would take many minutes to delete 100k items from the index.